hibernate-reactive-core/src/main/java/org/hibernate/reactive/context/impl/VertxContext.java

```diff
@@ -61,14 +63,18 @@ public void remove(Key<?> key) {
         }
     }
 
-    @Override
-    public void execute(Runnable runnable) {
-        io.vertx.core.Context context = vertxInstance.getVertx().getOrCreateContext();
-        if ( Vertx.currentContext() == context ) {
-            runnable.run();
-        }
-        else {
-            context.runOnContext( x -> runnable.run() );
-        }
-    }
+	@Override
+	public void execute(Runnable runnable) {
+		final io.vertx.core.Context currentContext = Vertx.currentContext();
```

Comment:
If there's an existing context we're running on, don't you think it would be preferreable to not mess with it?
I'd rather have your other assertions fail hard to spot such a case. (the assertions in Quarkus)

Reply:
I see. Yes, but here there is a risk to run on a root context. At least my assertion will catch this case.